recipes/isl/all/conanfile.py

```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
-            ])
-        if self.settings.compiler == "Visual Studio":
-            if tools.Version(self.settings.compiler.version) >= 15:
-                self._autotools.flags.append("-Zf")
-            if tools.Version(self.settings.compiler.version) >= 12:
-                self._autotools.flags.append("-FS")
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+            tc.configure_args.append("--with-gmp=system")
+            tc.configure_args.append(f'--with-gmp-prefix={unix_path(self, self.dependencies["gmp"].package_folder)}')
+        if self.settings.compiler == "msvc":
+            if Version(self.settings.compiler.version) >= 15:
+                tc.extra_cflags = ["-Zf"]
+            if Version(self.settings.compiler.version) >= 12:
+                tc.extra_cflags = ["-FS"]
+        env = tc.environment()
+        if self.settings.compiler == "msvc":
+            env.define("AR", f'{unix_path(self, self.dependencies["automake"].ar_lib)} lib')
+            env.define("CC", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo {self.settings.compiler.runtime}')
+            env.define("CXX", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo {self.settings.compiler.runtime}')
+            env.define("NM", "dumpbin -symbols")
+            env.define("OBJDUMP", ":")
+            env.define("RANLIB", ":")
+            env.define("STRIP", ":")
+        tc.generate(env)
 
     def build(self):
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        # Support building with source from Git reop
+        with chdir(self, self.source_folder):
+            command = "./autogen.sh"
+            if os.path.exists(command) and not os.path.exists("configure"):
+                self.run(command)
+        apply_conandata_patches(self)
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"), keep_path=False)
+        autotools = Autotools(self)
+        autotools.install()
 
         os.unlink(os.path.join(os.path.join(self.package_folder, "lib", "libisl.la")))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "isl"
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "ISL")
+        self.cpp_info.set_property("cmake_target_name", "ISL::ISL")
```

Comment:
```suggestion
        self.cpp_info.set_property("pkg_config_name", "isl")
```

there is no Find module or CMake config file upstream, do not dictate an unofficial name please

Reply:
Fixed